src/lib/comms/sol-coap.c

```diff
@@ -104,12 +104,19 @@ struct pending_reply {
 };
 
 struct outgoing {
+    struct sol_ptr_vector notified; /* Only for notification packets */
+    struct resource_context *context; /* Only set for notification packets */
```

Comment:
Wouldn't it be more direct if the context kept the reference to the notification packet?

But I still need to think about this a little more.